gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java

```diff
@@ -313,6 +321,11 @@ public List<ApiCriteria> computeApiCriteriaForUser(
                 apiCriteriaList.add(queryToCriteria(executionContext, apiQuery).ids(userGroupApiIds).build());
             }
 
+            if (!isBlank(apiQuery.getCategory())) {
+                var categoryName = categoryService.findById(apiQuery.getCategory(), executionContext.getEnvironmentId()).getKey();
```

Comment:
Searching in the apis tabele by category key, not category id, since apis table can category key and not category ids
<img width="1659" alt="Screenshot 2024-05-10 at 10 21 10" src="https://github.com/gravitee-io/gravitee-api-management/assets/8080303/3be99029-f96b-4e3a-bb02-4962d0d1b201">